Tests/Marketplace/upload_packs.py

```diff
@@ -995,7 +998,15 @@ def main():
             continue
 
         task_status = pack.format_metadata(user_metadata, index_folder_path, packs_dependencies_mapping, build_number,
-                                           current_commit_hash, pack_was_modified, statistics_handler)
+                                           current_commit_hash, pack_was_modified, statistics_handler, pack_names)
+
+        if pack.is_missing_dependencies:
+            # If the pack is dependent on a new pack (which is not yet in the index.json)
+            # we will note that it is missing dependencies.
+            # And finally after updating all the packages in index.json.
+            # We will go over the pack again to add what was missing
+            packs_missing_dependencies.append(pack)
+
```

Comment:
please return a flag like task_status
```suggestion
        if pack.is_missing_dependencies:
            # If the pack is dependent on a new pack (which is not yet in the index.zip as it might not have been iterated yet)
            # we will note that it is missing dependencies.
            # And finally after updating all the packages in index.zip - i.e. the new pack exists now.
            # We will go over the pack again to add what was missing.
            # See issue #37290
            packs_missing_dependencies.append(pack)

```